src/pages/index.js

```diff
@@ -41,41 +56,157 @@ const addCardPopup = new PopupWithForm(
 );
 addCardPopup.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarEditSubmit
+);
+editAvatarPopup.setEventListeners();
+
+const deleteConfirmation = new PopupWithConfirm("#delete-confirmation-modal");
+deleteConfirmation.setEventListeners();
+
 const cardPreviewPopup = new PopupWithImage("#image-modal");
 cardPreviewPopup.setEventListeners();
 
-const cardSection = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
-cardSection.renderItems(cardData);
+const user = new UserInfo({
+  name: ".profile__title",
+  job: ".profile__description",
+  avatar: ".profile__picture",
+});
 
-const user = new UserInfo(".profile__title", ".profile__description");
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "b7515436-5dfa-4f13-a09a-5183c1df5fb3",
+    "Content-Type": "application/json",
+  },
+});
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 function handleProfileEditSubmit(profileData) {
-  const title = profileData.title;
-  const description = profileData.description;
-  user.setUserInfo(title, description);
-  editProfilePopup.close();
+  console.log("profileData:", profileData);
+  const name = profileData.title;
+  const job = profileData.description;
+  console.log("name:", name, "job:", job); //debugging
+  profileEditSubmitBtn.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```